### PR TITLE
GEODE-8859: fix redis-compatability data structure bucket memory-usag…

### PR DESCRIPTION
**Pull request by jhutchison**
…e reporting

-implement default getForceReCalculateSize in RedisData interface
-expose dataStoreBytesInUse on GeodeRedisServer
-expose dataStoreBytesInUse on GeodeRedisService
-expose dataStoreBytesInUse on RedisClusterStartupRule

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Review by dschneider-pivotal — APPROVED** [review]

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/BucketSizeTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package org.apache.geode.redis.internal.data;
```
Comment (dschneider-pivotal) [inline geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/BucketSizeTest.java:1]: add the copyright header comment

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/ParitionedRegionStatsUpdateTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package org.apache.geode.redis.internal.data;
```
Comment (dschneider-pivotal) [inline geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/ParitionedRegionStatsUpdateTest.java:1]: add the copyright header comment

**Review by dschneider-pivotal — COMMENTED** [review]
two of your new files are missing the copyright header. That is why rat is failing in the build

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/ParitionedRegionStatsUpdateTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import redis.clients.jedis.Jedis;
+
+import org.apache.geode.test.awaitility.GeodeAwaitility;
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.dunit.rules.RedisClusterStartupRule;
+
+public class ParitionedRegionStatsUpdateTest {
```
Comment (sabbey37) [inline geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/ParitionedRegionStatsUpdateTest.java:17]: `Partitioned` is spelled incorrectly.

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/ParitionedRegionStatsUpdateTest.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    long initialDataStoreBytesInUse =
+        clusterStartUpRule.getDataStoreBytesInUseForDataRegion(server1);
+
+    for (int i = 0; i < 1000; i++) {
+      jedis1.append(KEY, LONG_APPEND_VALUE);
+    }
+
+    long finalDataStoreBytesInUse = clusterStartUpRule.getDataStoreBytesInUseForDataRegion(server1);
+    assertThat(initialDataStoreBytesInUse).isLessThan(finalDataStoreBytesInUse);
+  }
```
Comment (sabbey37) [inline geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/ParitionedRegionStatsUpdateTest.java:58]: This test seems like it's testing the same thing as the`should_show_increase_in_dataStore_bytes_in_use_given_data_added` test in `BucketSizeTest.java`.  Are they supposed to be testing different things?

**Review by sabbey37 — CHANGES_REQUESTED** [review]

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/ParitionedRegionStatsUpdateTest.java (hunk 4 above)
Comment (jhutchison) [reply geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/ParitionedRegionStatsUpdateTest.java:58]: hmm-- the file is supposed to be renamed-  I guess both are in there... I'll fix.  thanks

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/ParitionedRegionStatsUpdateTest.java (hunk 3 above)
Comment (jhutchison) [reply geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/ParitionedRegionStatsUpdateTest.java:17]: thx will fix

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/BucketSizeTest.java (hunk 1 above)
Comment (jhutchison) [reply geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/BucketSizeTest.java:1]: will add - thx

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/ParitionedRegionStatsUpdateTest.java (hunk 2 above)
Comment (jhutchison) [reply geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/ParitionedRegionStatsUpdateTest.java:1]: will add - thx

**Review by upthewaterspout — APPROVED** [review]
Looks good. At some point we need to implement Sizeable for RedisData, but I think this should work for now.

**Review by sabbey37 — APPROVED** [review]

**Pull request merged**